tracer/src/Datadog.Trace.Manual/Configuration/ImmutableTracerSettings.cs

```diff
@@ -0,0 +1,191 @@
+        CustomSamplingRules = GetValue<string?>(initialValues, TracerSettingKeyConstants.CustomSamplingRules, null);
+        Environment = GetValue<string?>(initialValues, TracerSettingKeyConstants.EnvironmentKey, null);
+        GlobalSamplingRate = GetValue<double?>(initialValues, TracerSettingKeyConstants.GlobalSamplingRateKey, null);
+        GlobalTags = GetValue<IReadOnlyDictionary<string, string>?>(initialValues, TracerSettingKeyConstants.GlobalTagsKey, null) ?? new ConcurrentDictionary<string, string>();
+        GrpcTags = GetValue<IReadOnlyDictionary<string, string>?>(initialValues, TracerSettingKeyConstants.GrpcTags, null) ?? new ConcurrentDictionary<string, string>();
+        HeaderTags = GetValue<IReadOnlyDictionary<string, string>?>(initialValues, TracerSettingKeyConstants.HeaderTags, null) ?? new ConcurrentDictionary<string, string>();
+        KafkaCreateConsumerScopeEnabled = GetValue(initialValues, TracerSettingKeyConstants.KafkaCreateConsumerScopeEnabledKey, true);
+        LogsInjectionEnabled = GetValue(initialValues, TracerSettingKeyConstants.LogsInjectionEnabledKey, false);
+        MaxTracesSubmittedPerSecond = GetValue(initialValues, TracerSettingKeyConstants.MaxTracesSubmittedPerSecondKey, 100);
+        ServiceName = GetValue<string?>(initialValues, TracerSettingKeyConstants.ServiceNameKey, null);
+        ServiceVersion = GetValue<string?>(initialValues, TracerSettingKeyConstants.ServiceVersionKey, null);
+        StartupDiagnosticLogEnabled = GetValue(initialValues, TracerSettingKeyConstants.StartupDiagnosticLogEnabledKey, true);
+        StatsComputationEnabled = GetValue(initialValues, TracerSettingKeyConstants.StatsComputationEnabledKey, false);
+        TraceEnabled = GetValue(initialValues, TracerSettingKeyConstants.TraceEnabledKey, true);
+        TracerMetricsEnabled = GetValue(initialValues, TracerSettingKeyConstants.TracerMetricsEnabledKey, false);
+
+#pragma warning disable CS0618 // Type or member is obsolete
+        Exporter = new ImmutableExporterSettings(this);
+        AnalyticsEnabled = GetValue(initialValues, TracerSettingKeyConstants.AnalyticsEnabledKey, false);
+#pragma warning restore CS0618 // Type or member is obsolete
+
+        Integrations = IntegrationSettingsHelper.ParseImmutableFromAutomatic(initialValues);
+
+        static T GetValue<T>(Dictionary<string, object?> results, string key, T defaultValue)
+        {
+            if (results.TryGetValue(key, out var value)
+             && value is T t)
+            {
+                return t;
+            }
+
+            return defaultValue;
+        }
+    }
+
+    /// <summary>
+    /// Gets the default environment name applied to all spans.
+    /// </summary>
+    [Instrumented]
+    public string? Environment { get; }
+
+    /// <summary>
+    /// Gets the Uri where the Tracer can connect to the Agent.
+    /// </summary>
+    [Instrumented]
+    internal Uri AgentUri { get; }
```

Comment:
The `ImmutableExporterSettings Exporter` property says to use the AgentUri property, so I think this is supposed to be public?

```suggestion
    public Uri AgentUri { get; }
```

Reply:
Good catch 👍